converter.go

```diff
@@ -354,6 +354,74 @@ func decimalToBigFloat(num decimal128.Num, scale int64) *big.Float {
 	return new(big.Float).Quo(f, s)
 }
 
+// ArrowSnowflakeTimestampToTime converts original timestamp returned by Snowflake to time.TIme
+func ArrowSnowflakeTimestampToTime(rec arrow.Record, rb *ArrowBatch, colIdx int, recIdx int) *time.Time {
+	scale := int(rb.scd.RowSet.RowType[colIdx].Scale)
+	dbType := strings.ToUpper(rb.scd.RowSet.RowType[colIdx].Type)
+	return arrowSnowflakeTimestampToTime(rec.Column(colIdx), getSnowflakeType(dbType), scale, recIdx, rb.loc)
+}
+
+func arrowSnowflakeTimestampToTime(
+	column arrow.Array,
+	sfType snowflakeType,
+	scale int,
+	recIdx int,
+	loc *time.Location) *time.Time {
+
+	if column.IsNull(recIdx) {
+		return nil
+	}
+	var ret time.Time
+	switch sfType {
+	case timestampNtzType:
+		if column.DataType().ID() == arrow.STRUCT {
+			structData := column.(*array.Struct)
```

Comment:
don't we need to check if the struct has all the fields to avoid errors when extracting the actual values?

Reply:
I'm not sure if this is needed. We know how the struct looks like for each timestamp type. Additional checking will be done for each row (can be time consuming) and will result in an error being returned. This would be SF issue that will come to us anyway.